src/dynamicprompts/parser/parse.py

```diff
@@ -413,6 +428,23 @@ def create_parser(
         parser_config=parser_config,
         prompt=variant_prompt,
     )
+
+    variable_ref = pp.Combine(
+        _configure_variable_access(
+            parser_config=parser_config,
+            prompt=pp.SkipTo(parser_config.variable_end),
+        ),
+    ).leave_whitespace()
+
+    # by default, the variable starting "${" and end "}" characters are
+    # stripped with variable access. This restores them so that the
+    # variable can be properly recognized at a later stage
+    variable_ref.set_parse_action(
+        lambda s, l, t: parser_config.variable_start
+        + "".join(t)
+        + parser_config.variable_end,
+    )
```

Comment:
Ruff is complaining about the naming here (and for good reason if you ask me), maybe
```suggestion
    variable_ref.set_parse_action(
        lambda string, location, token: f"{parser_config.variable_start}{''.join(token)}{parser_config.variable_end}",
    )
```
all in all?

Reply:
Changed.